Sources/Core/SoftwareBreakpointManager.cpp

```diff
@@ -94,23 +94,34 @@ ErrorCode SoftwareBreakpointManager::disableLocation(Site const &site,
   return kSuccess;
 }
 
-void SoftwareBreakpointManager::enable(Target::Thread *thread) {
-  super::enable(thread);
-
-  _enabled = true;
-}
-
-void SoftwareBreakpointManager::disable(Target::Thread *thread) {
-  super::disable(thread);
-
-  _enabled = false;
+void SoftwareBreakpointManager::afterResume() {
+  //
+  // Remove temporary breakpoints.
+  //
+  auto it = _sites.begin();
+  while (it != _sites.end()) {
+    it->second.type =
+        static_cast<Type>(it->second.type & ~kTypeTemporaryOneShot);
+    if (!it->second.type) {
+      // refs should always be 0 unless we have a kTypePermanent breakpoint.
+      DS2ASSERT(it->second.refs == 0);
+      _sites.erase(it++);
+    } else {
+      it++;
+    }
```

Comment:
You could rewrite this to be something like:
```
if (!it->second.type) {
  // blah
  DS2ASSERT(bloop);
  _sites.erase(it);
}
it++;
```

This feels cleaner than having `it++` twice. Thoughts?

Reply:
Yep, I meant to change that and forgot, good catch.

Reply:
Don't. After `_sites.erase(it)`, the iterator becomes invalid, so you can't do `it++`.

Reply:
😮

Reply:
Shows how much I know! :D